include/deal.II/base/exceptions.h

```diff
@@ -1020,6 +1020,18 @@ namespace StandardExceptions
  */
 namespace deal_II_exceptions
 {
+  namespace internals
+  {
+    /**
+     * Setting this variable to false will disable deal.II's exception mechanism
+     * to abort the problem. The Assert() macro will throw the exception instead
+     * and the AssertNothrow() macro will just print the error message. This
+     * variable should not be changed directly. Use disable_abort_on_exception()
+     * instead.
+     */
+    extern bool allow_abort_on_exception;
```

Comment:
We need another `extern` declaration for `show_stacktrace` here.

Reply:
Seems the declaration wasn't redundant. :wink:
Let me just remove the last commit. There is no reason for exposing `show_stacktrace`.

Reply:
whoops :)

Reply:
I guess that as long as the compiler is happy with multiple `extern` declarations we should keep it.

Reply:
@drwells I am not sure I understand if you want me to change anything or not.

Reply:
No, things are fine. I originally commented about the extra `extern` declarations because I thought the compiler would be upset if we did it.